### DISCUSSION TITLE
Is there a way to rewrite url for files inside the public directory

### DISCUSSION THREAD

**Discussion by thesobercoder** [vercel/next.js#20843] [Help] [answered]
I have an application where I'm hosting static Storybook files inside the public folder. Currently, I'm just redirecting it to something like 

```
<Link href="/docs/index.html"><a>Docs</a></Link>
```

 and it's working fine, though I'd like the users to see

```http://localhost:3000/docs``` 

instead of 

```http://localhost:3000/docs/index.html```. 

Is there a way I can achieve this with v10.0.1?

**Comment by ijjk** [comment, marked as answer]
Hi, you can leverage the rewrites config in `next.config.js` to achieve this https://nextjs.org/docs/api-reference/next.config.js/rewrites, note if you have a `pages/index.js` it will take priority over the rewrite.

**Reply by thesobercoder** [reply]
I already have the rewrite rule added. 

```
module.exports = withNx({
  async rewrites() {
    return [
      {
        source: '/index.html',
        destination: '/docs',
      },
    ];
  },
});
```

The problem is that when I compile storybook I couldn't find an option to rename the start file from `index.html` to something else. Though I must add that my `index.html` file is not inside the `pages` folder but `public` folder.

**Reply by ijjk** [reply]
If the `index.html` file is in your `public` directory and you want visiting `/docs` to load the `index.html` then the rewrite is backwards e.g.

```js
{
  source: '/docs',
  destination: '/index.html'
}
```

**Reply by ijjk** [reply]
Are these JS bundles in your public folder? The source should be the URL you expect it to be requested at and then the destination the location in the public folder or a route from the pages folder

**Reply by thesobercoder** [reply]
I figured it out. This is what I went with.

```
{
  source: '/:slug(runtime~main.*.bundle.js)',
  destination: '/docs/:slug',
}
```

**Reply by pke** [reply]
> ```js
> {
>   source: '/docs',
>   destination: '/index.html'
> }
> ```

This will break relative links inside `index.html`. For instance ´<a href="page2.html">` will point to the root "/page2.html" instead of "/docs/page2.html".

When rendering ´docs/index.html` directly then the link is correct.

**Reply by founderpad** [reply]
Do these rewrites work locally? They don't work for me :s
